### DISCUSSION TITLE
I need help with a Javascript string replacement boobytrap

### DISCUSSION THREAD

**Discussion by TexasGaidheal** [nextgenhealthcare/connect#5220] [Q&A] [answered]
I'm trying to replace all instances of a tab character in a string variable with a space character. I expected that to be simple, but my first attempt and several variations failed. A search of the support forum produced many interesting and useful posts about regular expressions but no solution to the problem. Here is the error message that I get, edited to reduce clutter:
```
44:   var tempStr = str.replace(/\t/g, ' ');

LINE NUMBER: 44
DETAILS: The choice of Java method java.lang.String.replace 
matching JavaScript argument types (function,string) is ambiguous; 

candidate methods are: 
    class java.lang.String replace(char,char)
    class java.lang.String replace(java.lang.CharSequence,java.lang.CharSequence)
```
This is in Mirth 4.0.1.

What is the best way to do this?

**Comment by pacmano1** [comment, marked as answer]
JS vs java variable types.

```JS
var stuff = '				tabs in front of this'
$c('stuff',stuff)
$c('this_stuff',stuff.replace(/\t/g,''))

var junk = new java.lang.String('			more tabs')
$c('junk',junk)
$c('this_junk',junk.replace('\t',''))

$c('this_junk1',junk.replace(/\t/g,''))   // DOES NOT WORK same error as yours
```

![image](https://user-images.githubusercontent.com/44065187/171281097-c54a107e-6579-44d0-8991-21ad204eb75c.png)

**Reply by mjtrac** [reply]
FWIW, also: 
```
$c('this_junk2',junk.replaceAll(/\t/,''));  
$c('this_stuff2',stuff.replaceAll(/\t/,''));  // TypeError: Cannot find function replaceAll in object ... tabs in front of this.

```

**Reply by TexasGaidheal** [reply]
Thanks pacmano1.

**Reply by tonygermano** [reply]
`var tempStr = String(str).replace(/\t/g, ' ')` Should also work to convert your java string to a js string before attempting the replace.
